src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -479,630 +577,1045 @@ public void LastIndexOf()
+            yield return new object[] { s_empty, 0, new[] { 2, 3, 4 } };
+            yield return new object[] { s_manyElements, 0, new int[0] };
+            yield return new object[] { s_empty, 0, s_empty };
+            yield return new object[] { s_empty, 0, s_oneElement };
+            yield return new object[] { s_oneElement, 0, s_empty };
+            yield return new object[] { s_empty, 0, new uint[] { 1, 2, 3 } };
+            yield return new object[] { s_manyElements, 0, new uint[] { 4, 5, 6 } };
+            yield return new object[] { s_manyElements, 3, new uint[] { 4, 5, 6 } };
+        }
 
-            Assert.Equal(list, list.RemoveRange(new[] { 5 }));
-            Assert.Equal(ImmutableArray.Create<int>(), ImmutableArray.Create<int>().RemoveRange(new[] { 1 }));
+        [Theory]
+        [MemberData(nameof(RemoveAtData))]
+        public void RemoveAt(IEnumerable<int> source, int index)
+        {
+            var array = source.ToImmutableArray();
+            var expected = source.Take(index).Concat(source.Skip(index + 1));
+            Assert.Equal(expected, array.RemoveAt(index));
+        }
 
-            var listWithDuplicates = ImmutableArray.Create(1, 2, 2, 3);
-            Assert.Equal(new[] { 1, 2, 3 }, listWithDuplicates.RemoveRange(new[] { 2 }));
-            Assert.Equal(new[] { 1, 3 }, listWithDuplicates.RemoveRange(new[] { 2, 2 }));
-            Assert.Equal(new[] { 1, 3 }, listWithDuplicates.RemoveRange(new[] { 2, 2, 2 }));
+        public static IEnumerable<object[]> RemoveAtData()
+        {
+            yield return new object[] { s_oneElement, 0 };
+            yield return new object[] { s_manyElements, 0 };
+            yield return new object[] { s_manyElements, 1 };
+            yield return new object[] { s_manyElements, 2 };
         }
 
-        [Fact]
-        public void RemoveRangeImmutableArrayTest()
+        [Theory]
+        [MemberData(nameof(Int32EnumerableData))]
+        public void RemoveAtInvalid(IEnumerable<int> source)
         {
-            var list = ImmutableArray.Create(1, 2, 3);
+            var array = source.ToImmutableArray();
 
-            ImmutableArray<int> removed2 = list.RemoveRange(ImmutableArray.Create(2));
-            Assert.Equal(2, removed2.Length);
-            Assert.Equal(new[] { 1, 3 }, removed2);
+            Assert.Throws<ArgumentOutOfRangeException>("index", () => array.RemoveAt(-1));
+            Assert.Throws<ArgumentOutOfRangeException>("length", () => array.RemoveAt(array.Length));
+            Assert.Throws<ArgumentOutOfRangeException>("index", () => array.RemoveAt(array.Length + 1));
+        }
 
-            ImmutableArray<int> removed13 = list.RemoveRange(ImmutableArray.Create(1, 3, 5));
-            Assert.Equal(1, removed13.Length);
-            Assert.Equal(new[] { 2 }, removed13);
+        [Theory]
+        [InlineData(-1, Skip = "TODO. See comment in ImmutableArray<T>.RemoveRange(int, int).")]
```

Comment:
tl;dr `RemoveRange` on a default array is throwing AOORE when the index is negative, but when other parameters are invalid it's throwing NRE because it evaluates `self.Length`. Can I fix this?